pkg/cmd/roachtest/tests/admission_control_snapshot_overload.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/spec"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/install"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/prometheus"
+)
+
+// This test sets up a 3-node CRDB cluster on 8vCPU machines, loads
+// it up with a large TPC-C dataset, and sets up a foreground load of kv95/1b.
+// The TPC-C dataset has a replication of factor of 1 and is used as the large
+// lump we bus around through snapshots -- snapshots that are send to the node
+// containing leaseholders serving kv95 traffic. Snapshots follow where the
+// leases travel, cycling through each node, evaluating performance isolation in
+// the presence of snapshots.
+//
+// TODO(irfansharif): This primarily stresses CPU; write an equivalent for IO.
+// TODO(irfansharif): This is a relatively long-running test (takes ~3.5hrs);
+// make it shorter.
+func registerSnapshotOverload(r registry.Registry) {
+	r.Add(registry.TestSpec{
+		Name:  "admission-control/snapshot-overload",
+		Owner: registry.OwnerAdmissionControl,
+		// TODO(irfansharif): After two weeks of nightly baking time, reduce
+		// this to a weekly cadence. This is a long-running test and serves only
+		// as a coarse-grained benchmark.
+		// 	Tags:    []string{`weekly`},
+		Cluster: r.MakeClusterSpec(4, spec.CPU(8)),
+		Run: func(ctx context.Context, t test.Test, c cluster.Cluster) {
+			if c.Spec().NodeCount < 4 {
+				t.Fatalf("expected at least 4 nodes, found %d", c.Spec().NodeCount)
+			}
+
+			c.Put(ctx, t.Cockroach(), "./cockroach", c.All())
+			crdbNodes := c.Spec().NodeCount - 1
+			workloadNode := crdbNodes + 1
+			for i := 1; i <= crdbNodes; i++ {
+				startOpts := option.DefaultStartOpts()
+				startOpts.RoachprodOpts.ExtraArgs = append(startOpts.RoachprodOpts.ExtraArgs, fmt.Sprintf("--attrs=n%d", i))
+				c.Start(ctx, t.L(), startOpts, install.MakeClusterSettings(), c.Node(i))
+			}
+
+			db := c.Conn(ctx, t.L(), crdbNodes)
+			defer db.Close()
+
+			// Set a replication factor of 1 and pin replicas to n1 by default.
+			// For the active dataset (we use kv further below) we'll use a
+			// different config.
+			t.Status(fmt.Sprintf("configuring default zone and settings (<%s)", 30*time.Second))
+			{
```

Comment:
Should this also be guarded by `SkipInit()`?

Reply:
Nah, only want to use --skip-init to avoid long running steps, not every step. If using skip-init, it's primarily for manual iteration, not really to have the test automatically set up a clean slate all the time. So I assume the person manually iterating knows what they're doing. In this test for ex. the second time I run it with --skip-init, data's actually placed differently from where it is. But that's fine since I know what's going on.